realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -74,7 +74,8 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
 
     // Column indices are cached to speed up opening typed Realm. If a Realm instance is created in one thread, creating
     // Realm instances in other threads doesn't have to initialize the column indices again.
-    private ColumnIndices typedColumnIndices;
+    private static final int MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY = 4;
```

Comment:
I am not quite convinced of caching old versions of `columnIndices`. Whenever we create a new realm instance, instead of getting a old schema cache, it will throw `MigrationException` first i think?